### DISCUSSION TITLE
batch_get_command_status doesn't return stderr when queued offline machine comes back and file does not exist

### DISCUSSION THREAD

**Discussion by jsdjjsd** [CrowdStrike/falconpy#883] [Q&A] [answered]
I'm using the batch_get_command class to schedule downloading of files from multiple machines over RTR. If a machine is online and the file does not exist it will return "Check your filename" as stderr. But if a machine is offline, and later comes back online, when using the batch_get_command_status class using batch_get_cmd_req_id from the initial request, the resulting stderr will stay empty. The only time the output of the batch_get_command_status request changes is if the file exists and has been downloaded and I get the sha256 field populated.

To reproduce this behaviour:
- Sleep target machine,
- Use batch_get_command to schedule a download of a file that does not exist on target machine,
- Wake target machine,
- Use batch_get_command_status and see that it never changes the output.

I would have thought that stderr would behave similarly as with batch_get_command, namely, return "Check your filename" if the file does not exist.

 - OS: Target machine is Windows 10, machine that runs Python 3 and FalconPy is Ubuntu 22.04,
 - Python: 3.10.6
 - FalconPy: 1.2.4

As far as I understand from the documentation this is the intended way to go about this, but is there another way this is intended to work?

**Comment by jshcodes** [comment]
Hi @jsdjjsd -

The process you describe above appears correct. We're recreating this issue on our side to confirm, and will reply back here once we have more information.

Thank you for reporting this!!! 😃

**Comment by jshcodes** [comment]
Hi @jsdjjsd -

We believe we've identified the issue. `BatchGetCmdStatus` is not getting results as the upload never completes. (Results are recorded upon successful completion of the **GET**.) 

If you instead try to use [`RTR_CheckCommandStatus`](https://www.falconpy.io/Service-Collections/Real-Time-Response.html#rtr_checkcommandstatus) (making sure to provide the individual task ID for this host) do you get the correct response?

**Comment by jsdjjsd** [comment, marked as answer]
Hi @jshcodes 

Thanks for looking into this. I've tested your suggestion now and at first I couldn't get `RTR_CheckCommandStatus` to return anything useful using the _session_id_ I get per resource from initiating the `BatchGetCmd`, but it seems that if I use _task_id_ instead I get something useful back, namely that while the endpoint is offline I get `complete: False` and when it comes back online it returns the _stderr_ I'm looking for. So if I just track the _task_id_ from respective resource I should be fine.

Thanks!

**Comment by jshcodes** [comment]
That's my bad for giving you the wrong ID type.  Updating the answer above for others. 😄
